src/uwtools/drivers/driver.py

```diff
@@ -174,43 +197,126 @@ def _taskname(self, suffix: str) -> str:
+    ):
         """
-        return "%s %s %s" % (cycle.strftime("%Y%m%d %HZ"), self._driver_name, suffix)
+        The driver.
 
-    def _taskname_with_cycle_and_leadtime(
-        self, cycle: datetime, leadtime: timedelta, suffix: str
-    ) -> str:
+        :param cycle: The cycle.
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param key_path: Keys leading through the config to the driver's configuration block.
         """
-        Returns a common tag for graph-task log messages.
+        super().__init__(config=config, dry_run=dry_run, cycle=cycle, key_path=key_path)
+        self._cycle = cycle
 
-        :param suffix: Log-string suffix.
+
+class AssetsCycleAndLeadtimeBased(Assets):
+    """
+    An abstract class to provision assets for cycle-and-leadtime-based components.
+    """
+
+    def __init__(
+        self,
+        cycle: datetime,
+        leadtime: timedelta,
+        config: Optional[Path] = None,
+        dry_run: bool = False,
+        key_path: Optional[list[str]] = None,
+    ):
         """
-        return "%s %s %s" % (
-            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S"),
-            self._driver_name,
-            suffix,
+        The driver.
+
+        :param cycle: The cycle.
+        :param leadtime: The leadtime.
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param key_path: Keys leading through the config to the driver's configuration block.
+        """
+        super().__init__(
+            config=config, dry_run=dry_run, cycle=cycle, key_path=key_path, leadtime=leadtime
         )
+        self._cycle = cycle
+        self._leadtime = leadtime
 
-    def _validate(self) -> None:
+
+class AssetsTimeInvariant(Assets):
+    """
+    An abstract class to provision assets for time-invariant components.
+    """
+
+    def __init__(
+        self,
+        config: Optional[Path] = None,
+        dry_run: bool = False,
+        key_path: Optional[list[str]] = None,
+    ):
         """
-        Perform all necessary schema validation.
+        The driver.
+
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param key_path: Keys leading through the config to the driver's configuration block.
         """
-        schema_name = self._driver_name.replace("_", "-")
-        validate_internal(schema_name=schema_name, config=self._config)
+        super().__init__(config=config, dry_run=dry_run, key_path=key_path)
 
 
 class Driver(Assets):
     """
     An abstract class for standalone component drivers.
     """
 
+    def __init__(
+        self,
+        config: Optional[Path] = None,
+        cycle: Optional[datetime] = None,
+        leadtime: Optional[timedelta] = None,
+        dry_run: bool = False,
+        batch: bool = False,
```

Comment:
`Driver` now provides its own constructor so that it can receive the `batch` argument, which `Assets` should not receive. It sets the `self._batch` instance variable from the formal argument so that `Assets` is relieved of that duty.